django/middleware/csrf.py

```diff
@@ -160,23 +160,47 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
-    def process_view(self, request, callback, callback_args, callback_kwargs):
-        if getattr(request, 'csrf_processing_done', False):
-            return None
-
+    def _get_current_token(self, request):
         try:
-            cookie_token = request.COOKIES[settings.CSRF_COOKIE_NAME]
+            if settings.CSRF_USE_SESSIONS:
+                session_token = request.session[settings.CSRF_SESSION_KEY]
+                csrf_token = _sanitize_token(session_token)
```

Comment:
What do you think about users acidentally overriding ``request.session['csrftoken']``? Should we care about this? I think sanitizing (e.g. dropping values of wrong lenth) makes sense in this case. Or raise ``SuspiciousOperation``?